test/test_util.jl

```diff
@@ -0,0 +1,188 @@
+using LinearAlgebra: AbstractTriangular
+
+#####
+##### We need to be able to convert everything into into a vector for use with FDM.jl
```

Comment:
It actually seems to me that all of this could go into FDM, since other packages may also want to use it. (Or alternatively FDM itself could use this machinery in order to be more permissive in what it accepts, then convert things to vectors internally as needed.)